src/engraving/libmscore/system.cpp

```diff
@@ -1804,6 +1804,52 @@ qreal System::lastNoteRestSegmentX(bool trailing)
     return margin;
 }
 
+//---------------------------------------------------------
+//   lastChordRest
+//    returns the last chordrest of a system for a particular track
+//---------------------------------------------------------
+
+ChordRest* System::lastChordRest(int track)
+{
+    for (auto measureBaseIter = measures().rbegin(); measureBaseIter != measures().rend(); measureBaseIter++) {
+        if ((*measureBaseIter)->isMeasure()) {
+            const Measure* measure = static_cast<const Measure*>(*measureBaseIter);
+            for (const Segment* seg = measure->last(); seg; seg = seg->prev()) {
+                if (seg->isChordRestType()) {
+                    ChordRest* cr = seg->cr(track);
+                    if (cr) {
+                        return cr;
+                    }
+                }
+            }
+        }
+    }
+    return nullptr;
+}
+
+//---------------------------------------------------------
+//   firstChordRest
+//    returns the last chordrest of a system for a particular track
+//---------------------------------------------------------
+
+ChordRest* System::firstChordRest(int track)
+{
+    qreal margin = score()->spatium();
+    for (const MeasureBase* mb : measures()) {
+        if (mb->isMeasure()) {
```

Comment:
if (!mb->isMeasure())

Reply:
Huh? In that case we'd be castring seomthing that isn't a measure into a measure??